impala/sqlalchemy.py

```diff
@@ -70,6 +88,26 @@ def post_create_table(self, table):
 class ImpalaTypeCompiler(GenericTypeCompiler):
     # pylint: disable=unused-argument
 
+    # https://docs.cloudera.com/documentation/enterprise/6/6.3/topics/impala_porting.html
+    # Impala only supports the STRING type
+    def visit_VARCHAR(self, type_):
+        return 'STRING'
+
+    visit_NCHAR = visit_VARCHAR
+    visit_VARCHAR = visit_VARCHAR
+    visit_NVARCHAR = visit_VARCHAR
+    visit_TEXT = visit_VARCHAR
+
+    # Impala only supports the TIMESTAMP type
+    def visit_DATETIME(self, type_):
+        return 'TIMESTAMP'
+
+    visit_TIME = visit_DATETIME
+
+    # Impala > 3.4.0 support the DATE type - https://issues.apache.org/jira/browse/IMPALA-6169
+    def visit_DATE(self, type_):
+        return 'DATE'
```

Comment:
@csringhofer Is there a way to "detect" the version of Impala being used within the context of this method?

If so; I'm thinking we could do something like:

```
if _impala_version > 3.4.0:
  return 'DATE'
else:
  return 'TIMESTAMP'
```

Reply:
We create the ImpalaTypeCompiler here: https://github.com/cloudera/impyla/blob/master/impala/sqlalchemy.py#L159
At that point we haven't connected to Impala yet, so we cannot know the exact version. I am also concerned making such behavior version dependent, because it makes it possible for an Impala version upgrade to brake an existing workflow.

A possible solution is to register two dialects in the future, e.g. impala and impala4, and impala4 could contain logic that only works on newer Impala versions.

Reply:
> A possible solution is to register two dialects in the future, e.g. impala and impala4, and impala4 could contain logic that only works on newer Impala versions.

I like this idea.  I'll modify the PR to just support TIMESTAMP types; with a TODO comment mentioning the need for a `impala4` dialect.

Reply:
See 
dfe9d00